Jenkinsfile

```
@@ -0,0 +1,191 @@
+          tool: checkStyle(pattern: 'target/checkstyle-result.xml')
+        )
+        recordIssues(
+          enabledForFailure: true,
+          blameDisabled: true,
+          forensicsDisabled: true,
+          qualityGates: [[threshold:60, type: 'TOTAL', unstable: false]],
+          tool: pmdParser(pattern: 'target/pmd.xml')
+        )
+      }
+        failure{
+          error "Test failure. Stopping pipeline execution!"
+        }
+        cleanup{
+          script{
+            sh 'sudo rm -rf target/'
+          }
+        }        
+      }
+    }
+
+    stage('Start RS-Proxy for Integration Testing'){
+      steps{
+        script{
+            sh 'scp src/test/resources/IUDX-Resource-Proxy-Server-Consumer-APIs.postman_collection_5.5.0.json jenkins@jenkins-master:/var/lib/jenkins/iudx/rs-proxy/Newman/'
+            sh 'docker compose -f docker-compose.test.yml up -d integTest'
+            sh 'sleep 30'
+        }
+      }
+      post{
+        failure{
+          script{
+            sh 'docker compose -f docker-compose.test.yml down --remove-orphans'
+          }
+        }
+      }
+    }
+    
+
+    stage('Integration Tests and OWASP ZAP pen test'){
+      steps{
+        node('built-in') {
+          script{
+            startZap ([host: '0.0.0.0', port: 8090, zapHome: '/var/lib/jenkins/tools/com.cloudbees.jenkins.plugins.customtools.CustomTool/OWASP_ZAP/ZAP_2.11.0'])
+            sh 'curl http://0.0.0.0:8090/JSON/pscan/action/disableScanners/?ids=10096'
+            sh 'HTTP_PROXY=\'127.0.0.1:8090\' newman run /var/lib/jenkins/iudx/rs-proxy/Newman/IUDX-Resource-Proxy-Server-Consumer-APIs.postman_collection_5.5.0.json -e /home/ubuntu/configs/rs-proxy-postman-env.json -n 2 --insecure -r htmlextra --reporter-htmlextra-export /var/lib/jenkins/iudx/rs-proxy/Newman/report/report.html --reporter-htmlextra-skipSensitiveData'
+            runZapAttack()
+            }
+        }
+      }
+      post{
+        always{
+          node('built-in') {
+            script{
+              archiveZap failHighAlerts: 1, failMediumAlerts: 1, failLowAlerts: 1
+            }  
+            publishHTML([allowMissing: false, alwaysLinkToLastBuild: true, keepAll: true, reportDir: '/var/lib/jenkins/iudx/rs-proxy/Newman/report/', reportFiles: 'report.html', reportTitles: '', reportName: 'Integration Test Report'])
+          }
+        }
+        failure{
+          error "Test failure. Stopping pipeline execution!"
+        }
+        cleanup{
+          script{
+            sh 'docker compose -f docker-compose.test.yml down --remove-orphans'
+          } 
+        }
+      }
+    }
+
+    stage('Continuous Deployment') {
+      when {
+        allOf {
+          anyOf {
+            changeset "docker/**"
+            changeset "docs/**"
+            changeset "pom.xml"
+            changeset "src/main/**"
+            triggeredBy cause: 'UserIdCause'
+          }
+          expression {
+            return env.GIT_BRANCH == 'origin/master';
```

branch is not master, it's main
```suggestion
            return env.GIT_BRANCH == 'origin/main';
```